lib/ansible/galaxy/collection/__init__.py

```diff
@@ -484,6 +486,7 @@ def install_collections(
                 concrete_artifacts_manager=artifacts_manager,
                 no_deps=no_deps,
                 allow_pre_release=allow_pre_release,
+                upgrade=upgrade,
             )
         except InconsistentCandidate as inconsistent_candidate_exc:
             # FIXME: Processing this error is hacky and should be removed along
```

Comment:
@s-hertel I think this is time to revert this. Mind adding `git revert 2dbb2ab59b0d79e42cec1e3f6b51634914d610a9`?

Reply:
Although, I'd probably make it the first commit in the branch. So you could instead do something like `git rebase -i origin/devel` and add `pick 2dbb2ab59b0d79e42cec1e3f6b51634914d610a9` in front of that TODO list.

Reply:
Sure, I was thinking pip-ish behavior and `--upgrade` were going to be separate PRs, but can do that here.

Reply:
Ah, I had an impression that these are related. But you could do a separate PR, sure.